08_project/alextmz/pkg/puppy/errors.go

```diff
@@ -0,0 +1,47 @@
+package puppy
+
+import "fmt"
+
+type Error struct {
+	Message string
+	Code    int
+}
+
+const (
+	ErrInvalidRequest = 400
+	ErrNotFound       = 404
+	ErrNegativeValue  = 499
```

Comment:
same comments as with lab 9

Reply:
I've refactored all of the error handling, became a bit more complex but it's now working the way I wanted with optional parameters and no http code references. I hope I didn't go too crazy on it.